assets/blocks/course-results-block/course-results-edit.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import CourseResultsSettings from './course-results-settings';
+import {
+	withColorSettings,
+	withDefaultColor,
+	withDefaultBlockStyle,
+} from '../../shared/blocks/settings';
+import { useCallback } from '@wordpress/element';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Edit course results block component.
+ *
+ * @param {Object}   props                         Component props.
+ * @param {string}   props.clientId                Block client ID.
+ * @param {string}   props.className               Custom class name.
+ * @param {Object}   props.mainColor               Header main color.
+ * @param {Object}   props.textColor               Header text color.
+ * @param {Object}   props.borderColor             Default border color.
+ * @param {Object}   props.defaultMainColor        Default main color.
+ * @param {Object}   props.defaultTextColor        Default text color.
+ * @param {Object}   props.defaultBorderColor      Default border color.
+ * @param {Object}   props.attributes              Block attributes.
+ * @param {boolean}  props.attributes.moduleBorder True if a modules should have a border.
+ * @param {Function} props.setAttributes           Block setAttributes callback.
+ */
+const CourseResultsEdit = ( props ) => {
+	const {
+		className,
+		defaultMainColor,
+		defaultTextColor,
+		defaultBorderColor,
+		textColor,
+		mainColor,
+		borderColor,
+		attributes: { moduleBorder },
+	} = props;
+
+	const styleRegex = /is-style-(\w+)/;
+	const style = className.match( styleRegex )?.[ 1 ];
+
+	// Minimal border element.
+	let minimalBorder;
+	if ( 'minimal' === style ) {
+		minimalBorder = (
+			<div className="wp-block-sensei-lms-course-results__module__separator" />
+		);
+	}
+
+	const renderSampleLesson = useCallback( ( lessonNumber ) => {
```

Comment:
Good idea! Switch in eee0d6ed063176b5b0379c29fee5128123cb2436. Do you think it is okay to keep them in the same file since they are just for use within this component?

Reply:
Cool! I think it's not a problem!

I was commenting about the JSDoc while you added it. haha. Another thing that we could do just to save some lines and indentation is to destructure the pros directly in the function argument, and the `sprintf ` could go directly to the render, so both components could return the render directly (saving an indentation level).

But not important at all! Just if you want to. haha

Reply:
I initially was restructuring in the function argument but for some reason I've seen more happening below (is it the safety with destructuring with `const`? Not sure). Anyway, since that doesn't really matter here I switched it back in e2d7029b9d96cbf92a6ac7f1ece97b44ba5b1ad7